frontend/src/qri/search.ts

```diff
@@ -0,0 +1,73 @@
+import { Dataset, NewDataset } from './dataset'
+
+export interface PageInfo {
+  nextUrl: string
+  page: number
+  pageSize: number
+  pageCount: number
+  prevUrl: string
+  resultCount: number
+}
+
+export interface SearchParams {
+  q: string
+  sort?: '' | 'name' | 'recentlyupdated'
```

Comment:
does this still need to be optional if the empty string is a permitted value? If you can make it required, let's do it.

Reply:
Yeah I'll make it required